types/src/validator_config.rs

```diff
@@ -35,20 +36,42 @@ impl MoveResource for ValidatorOperatorConfigResource {
 #[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
 pub struct ValidatorConfig {
     pub consensus_public_key: Ed25519PublicKey,
-    pub validator_network_address: RawEncNetworkAddress,
-    pub full_node_network_address: RawNetworkAddress,
+    /// This is an lcs serialized Vec<RawEncNetworkAddress>
+    pub validator_network_addresses: Vec<u8>,
+    /// This is an lcs serialized Vec<RawNetworkAddress>
+    pub full_node_network_addresses: Vec<u8>,
 }
 
 impl ValidatorConfig {
     pub fn new(
         consensus_public_key: Ed25519PublicKey,
-        validator_network_address: RawEncNetworkAddress,
-        full_node_network_address: RawNetworkAddress,
+        validator_network_addresses: Vec<u8>,
+        full_node_network_addresses: Vec<u8>,
     ) -> Self {
         ValidatorConfig {
             consensus_public_key,
-            validator_network_address,
-            full_node_network_address,
+            validator_network_addresses,
+            full_node_network_addresses,
         }
     }
+
+    /// Returns a filtered list of correct NetworkAddresses within the full_node_network_addresses
+    pub fn full_node_network_addresses<'a>(
+        &self,
+        err_callback: Option<Box<dyn Fn(String) + 'a>>,
```

Comment:
the callback seems anti-pattern to me, why not just map_err in the caller? (also why not deserialize to Vec\<RawNetworkAddress\> directly)

Reply:
what if I want to skip some because they aren't well formed? how do I note that?

Reply:
we probably should validate all on-chain resources are well-formed, so this can be deserialized directly without bothering

Reply:
that would be beautiful ... I'll improving this interface in a future commit -- as that would not be breaking.